lib/passport-configurator.js

```diff
@@ -491,6 +491,8 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
         passReqToCallback: true,
       }, options),
         function(req, profile, done) {
+          // Azure saml profile returns http://schemas.microsoft.com/identity/claims/objectidentifier
+          profile.id = profile.id || profile["http://schemas.microsoft.com/identity/claims/objectidentifier"];
```

Comment:
I am afraid add the alternative oid to profile for all saml format auth is too general, we can pass in a flag in options to specify this is an Azure profile, and assign the oid to profile.id when the flag is true.

```js
if (options.isAzureProfile) {
  const objIdentifier =  'http://schemas.microsoft.com/identity/claims/objectidentifier';
  profile.id = profile[objIdentifier];
}
```

Reply:
Hi @jannyHou,

We thought about it but realized it was already done before further in the code (https://github.com/GhizlaneA/loopback-component-passport/blob/852eac4224e0ad2ba51ec8fb15486983054ef6eb/lib/passport-configurator.js#L463)

```js
...
 function(req, iss, sub, profile, jwtClaims, accessToken, refreshToken,
 params, done) {
  // Azure openid connect profile returns oid
  profile.id = profile.id || profile.oid;
...
```

For us it makes sense as it does not break the code, but simply makes sure it works when using azure (which is the only provider I know of today, that does not return an id). Otherwise the value will be `undefined` which is not good anyway.

Do you think we should also edit that other part of the code or just keep it that way ?

Reply:
@cadesalaberry thanks! I see, then the approach in this PR makes sense, let's check it in since it's consistent with the code above.

While the lint complains the double quote:

> 11:12:28 /home/jenkins/workspace/nb/loopback-component-passport~master/7a48ad0e/lib/passport-configurator.js
11:12:28   495:46  error  Strings must use singlequote  quotes
11:12:28 
11:12:28 ✖ 1 problem (1 error, 0 warnings)

Please change it to single quote and I will help you land it, thanks!